src/main/java/spoon/support/reflect/declaration/CtTypeImpl.java

```diff
@@ -891,9 +892,16 @@ public String getQualifiedName() {
 	public Set<CtMethod<?>> getAllMethods() {
 		final Set<String> distinctSignatures = new HashSet<>();
 		final Set<CtMethod<?>> l = new SignatureBasedSortedSet<>();
```

Comment:
I would first merge this, with a clean implementation, and maybe a temporary behavioral change highlighted in the test diff. And then, we can concentrate on #1407.